src/core/exec-invoke.c

```diff
@@ -1216,17 +1216,9 @@ static int setup_pam(
                  * we'd never signal completion. */
                 exec_fd = safe_close(exec_fd);
 
-                /* Drop privileges - we don't need any to pam_close_session and this will make
-                 * PR_SET_PDEATHSIG work in most cases.  If this fails, ignore the error - but expect sd-pam
-                 * threads to fail to exit normally */
-
-                r = fully_set_uid_gid(uid, gid, /* supplementary_gids= */ NULL, /* n_supplementary_gids= */ 0);
-                if (r < 0)
-                        log_warning_errno(r, "Failed to drop privileges in sd-pam: %m");
-
                 (void) ignore_signals(SIGPIPE);
 
-                /* Wait until our parent died. This will only work if the above setresuid() succeeds,
+                /* Wait until our parent died. This will only work if the service process is privileged,
                  * otherwise the kernel will not allow unprivileged parents kill their privileged children
```

Comment:
Hmm? The previous comment is correct? The parent executor will drop privilege and become `systemd --user` instance eventually. So sd-pam process canmot be killed?

Maybe, also switch to PidRef for tracking the parent process. This allows us to exit gracefully.

Reply:
> Hmm? The previous comment is correct? The parent executor will drop privilege and become `systemd --user` instance eventually. So sd-pam process canmot be killed?

I am not familiar with the region, and I am not sure the original comment is correct or not...
But at least this makes pam_systemd successfully call ReleaseSession DBus method.

> Maybe, also switch to PidRef for tracking the parent process. This allows us to exit gracefully.

:+1: